tabbycat/results/models.py

```diff
@@ -376,3 +377,61 @@ def clean(self):
         if self.ballot_submission.debate != self.debate_team.debate:
             raise ValidationError(_("The ballot submission and debate team must "
                     "relate to the same debate."))
+
+
+class ScoreCriterion(models.Model):
+    """Score criterion for speaker score"""
+    tournament = models.ForeignKey(Tournament, models.CASCADE,
+        verbose_name=_("tournament"))
+    name = models.CharField(max_length=20,
+        verbose_name=("name"))
+    seq = models.IntegerField(verbose_name=_("sequence"))
+    weight = models.FloatField(verbose_name=_("weight"))
+    min_score = ScoreField(verbose_name=_("minimum score"))
+    max_score = ScoreField(verbose_name=_("maximum score"))
+    step = models.FloatField(verbose_name=_("step"))
+    required = models.BooleanField(default=True,
+        verbose_name="required")
+
+    class Meta:
+        unique_together = [('tournament', 'seq')]
+        verbose_name = _("score criterion")
+        verbose_name_plural = _("score criteria")
+
+    def __str__(self):
+        return ("{0.name} at {0.tournament}").format(self)
+
+
+class SpeakerCriterionScore(models.Model):
+
+    score = ScoreField(verbose_name=_("score"))
+    criterion = models.ForeignKey(ScoreCriterion, models.CASCADE,
+        verbose_name=_("score criterion"))
+    speaker_score = models.ForeignKey(SpeakerScore, models.CASCADE,
+        verbose_name="speaker score")
+
+    class Meta:
+        unique_together = [('speaker_score', 'criterion')]
+        verbose_name = _("speaker score for criterion")
+        verbose_name_plural = _("speaker scores for criteria")
+
+    def __str__(self):
+        return ("[{0.speaker_score_id}/{0.id}] {0.score}").format(self)
+
+
+class SpeakerCriterionScoreByAdj(models.Model):
+
+    score = ScoreField(verbose_name=_("score"))
+    criterion = models.ForeignKey(ScoreCriterion, models.CASCADE,
+        verbose_name=_("score criterion"))
+    speaker_score_by_adj = models.ForeignKey(SpeakerScoreByAdj, models.CASCADE,
+        verbose_name="speaker score")
+
+    class Meta:
+        unique_together = [('speaker_score_by_adj', 'criterion')]
+        verbose_name = _("speaker score for criterion by adjudicator")
+        verbose_name_plural = _("speaker scores for criteria by adjudicator")
+
+    def __str__(self):
+        return ("[{0.ballot_submission_id}/{0.id}] {0.score} at {0.position} for "
```

Comment:
This one won't work as `ballot_submission_id`, `position`, `debate_team`, and `debate_adjudicator` aren't defined.